tests/unit/item_test.c

```diff
@@ -113,41 +113,245 @@ static void test_list_compare(void)
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("\\\\:blah", ':');
+    AppendItem(&expected, "\\", NULL);
+    AppendItem(&expected, "blah", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("\\\\\\:", ':');
+    AppendItem(&expected, "\\:", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("\\\\\\:blah", ':');
+    AppendItem(&expected, "\\:blah", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("blah\\\\:", ':');
+    AppendItem(&expected, "blah\\", NULL);
+    AppendItem(&expected, "", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("\\\\:\\\\", ':');
+    AppendItem(&expected, "\\", NULL);
+    AppendItem(&expected, "\\", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+
+    actual = SplitString("blah:\\\\:blue", ':');
+    AppendItem(&expected, "blah", NULL);
+    AppendItem(&expected, "\\", NULL);
+    AppendItem(&expected, "blue", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("blah:\\\\\\:blue", ':');
+    AppendItem(&expected, "blah", NULL);
+    AppendItem(&expected, "\\:blue", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("blah\\\\blue", ':');
+    AppendItem(&expected, "blah\\blue", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    /* End string with backslash, danger of buffer overrun! */
+
+    actual = SplitString("blah\\", ':');
+    AppendItem(&expected, "blah\\", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
+    actual = SplitString("\\", ':');
+    AppendItem(&expected, "\\", NULL);
+    assert_true(ListsCompare(actual, expected));
+    DeleteItemList(actual);   actual   = NULL;
+    DeleteItemList(expected); expected = NULL;
+    test_progress();
+
```

Comment:
Please include a string whose last character (after escaping) before a separator is a backslash.
Of course, that means the separator is preceded by an even number of backslashes; so the C source contains a multiple of four backslashes, followed by the separator.